atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -26,23 +31,39 @@
 class OracleTableNameUnmapper {
     private final ConnectionSupplier conns;
 
+    private LoadingCache<String, String> unmappingCache = CacheBuilder.newBuilder().build(
+            new CacheLoader<String, String>() {
+                @Override
+                public String load(String fullTableName) throws Exception {
+                    AgnosticResultSet results = conns.get().selectResultSetUnregisteredQuery(
```

Comment:
The `ConnectionSupplier` and `SqlConnection` interface is a resource seive. You could do your own connection management using `conns.getFresh()` and ensure closure of that one, basically what's done in `com.palantir.atlasdb.keyvalue.dbkvs.impl.BatchedDbReadTable#run`.